apps/meteor/client/views/admin/rooms/RoomsTable.tsx

```diff
@@ -16,41 +15,43 @@ import {
 } from '../../../components/GenericTable';
 import { usePagination } from '../../../components/GenericTable/hooks/usePagination';
 import { useSort } from '../../../components/GenericTable/hooks/useSort';
+import type { SearchFilters } from '../../../contexts/SearchFilterContext';
+import { SearchFilterContext } from '../../../contexts/SearchFilterContext';
 import RoomRow from './RoomRow';
 import RoomsTableFilters from './RoomsTableFilters';
 
-type RoomFilters = {
-	searchText: string;
-	types: OptionProp[];
-};
-
 const DEFAULT_TYPES = ['d', 'p', 'c', 'l', 'discussions', 'teams'];
 
 const RoomsTable = ({ reload }: { reload: MutableRefObject<() => void> }): ReactElement => {
 	const t = useTranslation();
 	const mediaQuery = useMediaQuery('(min-width: 1024px)');
 
-	const [roomFilters, setRoomFilters] = useState<RoomFilters>({ searchText: '', types: [] });
+	const { searchFilters } = useContext(SearchFilterContext);
 
-	const prevRoomFilterText = useRef<string>(roomFilters.searchText);
+	const prevRoomFilters = useRef<SearchFilters>(searchFilters);
 
 	const { sortBy, sortDirection, setSort } = useSort<'name' | 't' | 'usersCount' | 'msgs' | 'default' | 'featured'>('name');
 	const { current, itemsPerPage, setItemsPerPage, setCurrent, ...paginationProps } = usePagination();
-	const searchText = useDebouncedValue(roomFilters.searchText, 500);
+	const searchText = useDebouncedValue(searchFilters.searchText, 500);
 
 	const query = useDebouncedValue(
 		useMemo(() => {
-			if (searchText !== prevRoomFilterText.current) {
+			const filtersChanged =
```

Comment:
Just checking if it's good that when prevRoomFilters & searchFilters have different order but same keys, this to be `true`